### PR TITLE
feat: Add stash count to git_status module

### LINKED ISSUE #595: Git Status: Stash Count
#### Is your feature request related to a problem? Please describe.
Ability to see the number of stashes in a git folder. First suggested here https://github.com/starship/starship/issues/423#issuecomment-534896545 by @etiennemabille.

#### Describe the solution you'd like
New option `stash_count` with enabled and style attributes. When enabled shows a count after the stash symbol. e.g. `$3`

### PR DESCRIPTION
**Pull request by marblenix**
Added stash count to the git status module.

#### Description
The git_module has been updated to add the number of stashes for the current repo after the stash symbol ("$"). Documentation has been updated accordingly.

#### Motivation and Context
Closes #595

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):

image

#### How Has This Been Tested?
Tested on mac by running new unit tests and running the docker test. Also built a release build and manually tested that.
- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

---

I'm new to rust so please correct me if I made any obvious mistakes! I'll update the PR as soon as I can.

### CODE REVIEW

File: src/modules/git_status.rs [hunk 1]
```diff
@@ -52,9 +56,6 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         log::trace!("No stash object found");
     }
 
-    let repo_status = get_repo_status(&repository);
-    log::debug!("Repo status: {:?}", repo_status);
-
```
Comment (marblenix) [inline src/modules/git_status.rs]: I had to move this up to the first part in the file because of the immutable borrows above it. Please let me know if there's a better way to go about this.

File: src/modules/git_status.rs [hunk 2]
```diff
@@ -114,7 +115,15 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
 
     // Add the stashed segment
     if stash_object.is_ok() {
-        module.create_segment("stashed", &config.stashed);
```
Comment (brianlow) [inline src/modules/git_status.rs]: Do we still need the `stash_object` variable (used one line above)?

File: tests/testsuite/git_status.rs [hunk 3]
```diff
@@ -259,33 +259,54 @@ fn doesnt_show_untracked_file_if_disabled() -> io::Result<()> {
 }
 
 #[test]
-#[ignore]
```
Comment (brianlow) [inline tests/testsuite/git_status.rs]: Keep the `[ignore]` and add to new tests according to this comment: https://github.com/starship/starship/pull/434#discussion_r329673883

These tests can be run specifically with `cargo test git_status -- --ignored`

File: src/modules/git_status.rs (hunk 2 above)
Comment (marblenix) [reply src/modules/git_status.rs]: It doesn't appear so. Without it all tests pass and the stash module seems unaffected overall.

File: tests/testsuite/git_status.rs (hunk 3 above)
Comment (marblenix) [reply tests/testsuite/git_status.rs]: Updated with `#[ignore]` back in place.

For the new tests, were there any edge cases you'd like me to explore further than what I did cover?

Comment (brianlow) [reply tests/testsuite/git_status.rs]: Tests look good 👍🏼

File: src/modules/git_status.rs [hunk 4]
```diff
@@ -244,6 +247,17 @@ fn is_untracked(status: Status) -> bool {
...
 }
 
+fn stashed_count(repository: &mut Repository) -> usize {
+    let mut count = 0;
+    repository
+        .stash_foreach(|_, _, _| {
+            count += 1;
+            true
+        })
+        .unwrap();
```
Comment (matchai) [inline src/modules/git_status.rs]: This `.unwrap()` could be a little dangerous here.
Ideally we would return an `Option` or `Result`, and early return the module as `None` if this is an `Error`.

**Comment by matchai** [comment]
@all-contributors Please add @marblenix for code and test. 🎉
